frontend/src/components/explorer/gemBrowser/Gene.vue

```diff
@@ -1,88 +1,46 @@
-                      </td>
-                      <td v-else> - </td>
-                    </tr>
-                  </table>
-                </div>
-                <ExtIdTable :type="type" :external-dbs="gene.externalDbs"></ExtIdTable>
-              </div>
-              <div class="column is-narrow">
-                <router-link v-if="model && gene.id" class="button is-info is-fullwidth is-outlined"
-                             :to="{ name: 'interaction', params: { model: model.short_name, id: gene.id } }">
-                  <span class="icon"><i class="fa fa-connectdevelop fa-lg"></i></span>&nbsp;
-                  <span>{{ messages.interPartName }}</span>
-                </router-link>
-                <br>
-                <maps-available :id="geneId" :type="type" :viewer-selected-i-d="gene.id" />
-                <gem-contact :id="geneId" :type="type" />
-              </div>
-            </div>
-            <reaction-table v-if="model && geneName" :source-name="geneName" :type="type" :selected-elm-id="geneName" />
-          </div>
-        </div>
-      </div>
-    </div>
-  </div>
+  <component-layout
+    :component-type="type" :component-name="gene.geneName"
+    :external-dbs="gene.externalDbs" query-component-action="genes/getGeneData"
+    :interaction-partner="true" :viewer-selected-i-d="gene.id"
+  >
+    <template v-slot:table>
+      <table v-if="gene && Object.keys(gene).length !== 0" class="table main-table is-fullwidth">
+        <tr v-for="el in mainTableKey" :key="el.name">
+          <td v-if="'display' in el"
+              class="td-key has-background-primary has-text-white-bis"
+              v-html="el.display"></td>
+          <td v-else-if="el.name === 'id'"
+              class="td-key has-background-primary has-text-white-bis">
+            {{ model? model.short_name : '' }} ID
+          </td>
+          <td v-else
+              class="td-key has-background-primary has-text-white-bis">{{ reformatTableKey(el.name) }}</td>
+          <td v-if="gene[el.name]">
+            <span v-if="'modifier' in el" v-html="el.modifier(gene)">
+            </span>
+            <span v-else>
+              {{ gene[el.name] }}
+            </span>
+          </td>
+          <td v-else> - </td>
+        </tr>
+      </table>
+    </template>
+  </component-layout>
 </template>
 
+
 <script>
 import { mapGetters, mapState } from 'vuex';
-import NotFound from '@/components/NotFound';
-import Loader from '@/components/Loader';
-import ExtIdTable from '@/components/explorer/gemBrowser/ExtIdTable';
-import MapsAvailable from '@/components/explorer/gemBrowser/MapsAvailable';
-import ReactionTable from '@/components/explorer/gemBrowser/ReactionTable';
-import GemContact from '@/components/shared/GemContact';
+import ComponentLayout from '@/layouts/explorer/gemBrowser/ComponentLayout';
 import { generateSocialMetaTags, reformatTableKey } from '@/helpers/utils';
 import { default as messages } from '@/helpers/messages';
```

Comment:
This, along with `messages,` in `data` should be removed. This also applies for `Metabolite.vue`